File: API/Models/Vehicle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore.Metadata.Internal;

namespace API.Models
{
    public class Vehicle
    {
        public int Id { get; set; }
        public string Category { get; set; }
        public string Manufacturer { get; set; }
        public string Model { get; set; }
        public string YearOfManufacture { get; set; }
        public string RegistrationPlate { get; set; }
        public DateTime DateOfRegistration { get; set; }
        public DateTime DateOfExpiration { get; set; }
        public bool IsRegistered { get; set; }
        public Owner Owner { get; set; }
        public int RegistrationNotificationId { get; set; }
        public RegistrationNotification RegistrationNotification { get; set; }
        public List<MaintenanceNotification> MaintenanceNotifications { get; set; }
    }
}

File: client/src/app/models/vehicle.ts
export interface Vehicle {
  id: number
  category: string
  manufacturer: string
  model: string
  yearOfManufacture: string
  registrationPlate: string
  dateOfRegistration: string
  dateOfExpiration: string
  isRegistered: boolean
  numberOfNotifications: number
  ownerId: number
}

export interface VehicleParams {
  searchTerm?: string;
  categories?: string[];
}